public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.tsx

```diff
@@ -74,6 +78,17 @@ export const LokiQueryBuilder = React.memo<Props>(({ datasource, query, onChange
     return undefined;
   }, [query]);
 
+  useEffect(() => {
```

Comment:
In Loki query builder (in contrast to PrometheusQueryBuilder), we want to get hints from sampledData, not received data. That's why we are fetching it here).